server/src/main/java/org/apache/lucene/index/BaseStarTreeBuilder.java

```diff
@@ -0,0 +1,648 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.apache.lucene.index;
```

Comment:
Sorry did we discuss the package name structure?

Reply:
This was done intentionally. In order to write doc values for dimensions and metrics, we need the `SortedNumericDocValuesWriter` implementation, which is package private. 
You can check the whole end-to-end implementation over [here](https://github.com/opensearch-project/OpenSearch/compare/main...sarthakaggarwal97:OpenSearch:startree-file-formats-codec-merge) for reference

Reply:
Update: Will move the BaseStarTreeBuilder to startree package. Will have a helper over `org.apache.lucene.index` to help us with SortedNumericDocValuesWriter.